### ISSUE TITLE
Allow launching additional Alacritty instances/windows in an empty workspace Mac OS

### ISSUE THREAD

**Issue by ethanculler** [alacritty/alacritty#4230]
I'm not sure if this would be considered a bug report or a feature request, but i'm going to assume it's a new feature. I, generally speaking, compartmentalize the resources for a specific task within the same "workspace" (I guess Apple calls them "Desktops"). This means that when I am forced to context switch from an incomplete task to a new task my workflow tends to be:
- Switch to an empty Desktop
- create new window/instance of terminal
- navigate to source related to new task
- launch editor (or whatever needs to be done)

This currently does not work with the version of alacritty i'm running, and couldn't find any other issues around it. I can launch new windows of alacritty if i'm currently focused on an existing alacritty window, but I can't figure out a way to launch a new window directly into a Desktop which does not already have an instance.

Steps to reproduce:
Path 1.
- open alacritty in a MacOS desktop
- switch to a new desktop
- Cmd-tab to "focus" alacritty (make sure you have the "when switching to an application, switch to a space with open windows for the application" setting disabled)
- Attempt to launch a new window with Cmd-n

Path 2.
- open alacritty in a MacOS desktop
- switch to a new desktop
- attempt to launch new alacritty instance from spotlight or applications.

### System

OS: macOS Version 10.15.6 (19G2021)
Version: alacritty 0.5.0 (a6681e3)
### Logs

Crashes: NA
Font/Terminal size:
Created log file at "/var/folders/yp/_95xghk950j22p6ykhz24xj80000gr/T/Alacritty-85903.log"
[2020-09-14 10:17:26.622738000] [DEBUG] Using environment locale: en_US.UTF-8
[2020-09-14 10:17:26.623009000] [INFO] Welcome to Alacritty
[2020-09-14 10:17:26.623080000] [INFO] No configuration file found
[2020-09-14 10:17:26.671916000] [DEBUG] Estimated DPR: 2
[2020-09-14 10:17:26.672025000] [DEBUG] Estimated Cell Size: 13 x 26
[2020-09-14 10:17:26.672099000] [DEBUG] Estimated Dimensions: None
[2020-09-14 10:17:26.735423000] [INFO] Device pixel ratio: 2
[2020-09-14 10:17:26.740458000] [INFO] Initializing glyph cache...
[2020-09-14 10:17:26.802041000] [INFO] ... finished initializing glyph cache in 0.0613774s
[2020-09-14 10:17:26.802138000] [INFO] Cell Size: 13 x 26
[2020-09-14 10:17:26.802189000] [INFO] Padding: 0 x 0
[2020-09-14 10:17:26.802238000] [INFO] Width: 1600, Height: 1200
[2020-09-14 10:17:26.802558000] [INFO] PTY dimensions: Line(46) x Column(123)
[2020-09-14 10:17:26.808410000] [INFO] Initialisation complete
[2020-09-14 10:17:27.420301000] [DEBUG] [unhandled osc_dispatch]: [['1',],['~',],] at line 785
[2020-09-14 10:17:32.708288000] [INFO] Goodbye
Deleted log file at "/var/folders/yp/_95xghk950j22p6ykhz24xj80000gr/T/Alacritty-85903.log"
Keyboard and bindings: `alacritty --print-events`

**Comment by chrisduerr** [comment]
This is a duplicate of #607.

**Comment by ethanculler** [comment]
Ah thanks. I actually read through that issue and thought that it was different based on the description of the features. I take it that this use case would be handled under a general new window flow.

Cheers!

**Comment by ethanculler** [comment]
@brightpuddle thanks for flagging that! Yes, I would like to. I see that there are some comments in [#607](https://github.com/alacritty/alacritty/issues/607) which are asking for similar workflows, but the workflow in this one definitely still doesn't work (confirmed in 0.10.1 as well just to be sure)

**Comment by brightpuddle** [comment]
To be clear, I'm not a maintainer of Alacritty. Just a fellow user struggling with a similar workflow. I'm assuming as the creator of the issue you'll have to reopen it, or maybe @chrisduerr can as a maintainer.

Although not a great workaround, if you happen to have something at a global level that can manage key maps, e.g. skhd, launchbar, alfred, etc, you could use that to launch an entirely new instance of Alacritty. I use yabai so already had skhd, and tested this by launching the binary directly (`/Applications/Alacritty.app/Resources/MacOS/alacritty`), and it works. This launches an *entirely* new instance of Alacritty though, so doesn't allow us to leverage the existing running instance and negates the benefits of #607. Maybe it actually makes sense to have an entirely different instance per workspace, but either way the current behavior is unexpected and doesn't seem to be working properly.

**Comment by kchibisov** [comment]
So you can't do `alacritty msg create-window` on that particular workspace to bring window? I don't quite understand the issue to be honest.

**Comment by brightpuddle** [comment]
`alacritty msg create-window` does indeed reuse the window. I wasn't aware of that, so thanks for that. I swapped out my skhd mapping and this works for me, and is probably a perfectly fine workaround since I'm always running skhd.

The issue is that a mapping within Alacritty doesn't work even with Alacritty being the foreground application. I have this mapping in my config: `  - {key: N, mods: Command, action: CreateNewWindow}`. It works fine when there's an Alacritty window focused, but when on a desktop without an Alacritty window with Alacritty still selected as the current app, pressing cmd-N does nothing. Pressing cmd-Q in the same situation closes Alacritty so Alacritty is still responding in some way, but it doesn't seem to be responding to custom mappings.

**In short:** Alacritty title bar is active, workspace has no Alacritty windows, Cmd-N mapping in place, Cmd-N doesn't work.

**Comment by kchibisov** [comment]
If you try binding, but without modifiers, will it work?

**Comment by brightpuddle** [comment]
No, still doesn't work. I bound it to just `n` (just removed the Command modifier. The mapping worked if I had an active Alacritty window focused, but not on a workspace without an active window.

**Comment by ethanculler** [comment]
@kchibisov separating out the key binding aspects of the issue for a moment. From a use case level there is currently no way (except perhaps the skhd work around) to open a new alacritty window into an active workspace which does not already have one.

In a MacOS world, most applications support up to 3 different ways to accomplish this.
1. switch to any empty desktop, select the application (e.g. cmd-tab until it is active), cmd-n for a new window. 
This is the pattern which @brightpuddle has flagged as still not working without use of skhd)
2. switch to any empty desktop, select the application (e.g. cmd-tab until it is active), use the file menu to create a new window
alacritty has no file menu, so this doesn't work
3. switch to any empty desktop, right click the running application in the dock, select "new window" option
there's no option to create a new window in this menu either.

I'll see if I can get skhd approved by my security team so I can try to test that work around, but it would definitely be nice if one of these 3 options (of course the first 1 would be preferred) worked out of the box.

Is it possible to re-open this as a feature request? I don't see a way to do so.

**Comment by walkie** [comment]
@chrisduerr Would it be possible to re-open this issue?

@brightpuddle's [comment here](https://github.com/alacritty/alacritty/issues/4230#issuecomment-1116706911) and @ethanculler's [comment directly above this one](https://github.com/alacritty/alacritty/issues/4230#issuecomment-1118017200) describe how this issue is not resolved by the fix to #607.

**Comment by cballenar** [comment]
Just to complement for anyone else dealing with this issue and using skhd as a workaround:

You can set a shortcut to 

1. create a new window in Alacritty and
2. if that doesn't work (if the app is closed), open the app as follows:

```
hyper - a : alacritty msg create-window || open /Applications/Alacritty.app
```

This will get you the standard behavior of other apps.

**Comment by swastkk** [comment]
Still facing this issue!

**Comment by bzadon** [comment]
Can we re-open this ticket? The "fix" is not working for me for version 0.14.0 (1) on macos

**Comment by nixpulvis** [comment]
I think we don't support this unfortunately. See https://github.com/alacritty/alacritty/issues/5891#issuecomment-1041339397

**Comment by nixpulvis** [comment]
I'm also in favor of re-opening this issue, but since it's a macOS thing it's less likely to be.

I used to be more motivated to push changes, but lately the project has become much more hostile to macOS and changes in general, so I've kinda backed off and lost touch with things a bit.

**Comment by adonh** [comment]
> > The title bar still shows Aclaritty. So doesn't that mean the application still has "focus"? 
> 
> Interesting, yes it does. But it doesn't mean that there is a focused window, which is what we currently need to receive events if my memory is correct. Also, I'm curious how you manage to get into that state. I guess it's possible with multiple monitors only?

For me, it's easy to get in this state. If you go to a macos desktop workspace where there isn't an Aclaritty terminal window open, Alt-Tab to select the app, then the app is active with no window on the monitor visible. If this were macos Terminal, Cmd-N would simply open a new window.

**Comment by nixpulvis** [comment]
alt-tab? I assume you mean command-tab.

That's interesting. For me it will then switch desktops to the one with the existing alacritty window.

What version of macOS are you running? I'm running 14.6.1.

**Comment by walkie** [comment]
@nixpulvis There's a System Setting "Desktop & Dock > Mission Control > When switching to an application, switch to a Space with open windows for the application". If that is disabled, then you won't switch spaces on cmd-tab.

In my experience, it's quite common for developers in particular to have this setting disabled, since typically you associate spaces with tasks or projects, and not with applications. This basically makes spaces work like multiple desktops on Linux. When that setting is disabled, having the application active but no window in focus is very common.

It's frustrating that this doesn't work properly, but even more frustrating that it's not even acknowledged as a bug. It's literally the only application I've encountered where Cmd-N doesn't work when the application is active but no window is present on the current space/desktop. It is absolutely standard behavior on macOS.

I'm currently using the [skhd](https://github.com/koekeishiya/skhd) workaround described above in this thread. That works most of the time but seems to break semi-regularly on updates. It's also the only thing I use skhd for, which requires elevated permissions, so I'd prefer not to rely on it for such a basic use case.

**Comment by nixpulvis** [comment]
OK, then I would personally be in favor of reopening this. But we'd have to see what @chrisduerr or @kchibisov have to say.

I might also try to dig into the code at some point, but I'm worried it's mixed up in winit as well.

**Comment by pean** [comment]
> It's frustrating that this doesn't work properly, but even more frustrating that it's not even acknowledged as a bug. It's literally the only application I've encountered where Cmd-N doesn't work when the application is active but no window is present on the current space/desktop. It is absolutely standard behavior on macOS.

Completely agree. This is the one blocker preventing me to switch to alacritty.

**Comment by walkie** [comment]
Here's another instance of this issue being closed as "works as intended": #7265
